server/src/search/search.controller.ts

```diff
@@ -89,49 +89,53 @@ const fullCourseSearch = async ({ search }: SearchQueryType) => {
   return fullSearch;
 };
 
+const courseSlicing = (sorted, searchType: string) => {
+  if (sorted && searchType === "search" && sorted.length > 5) {
+    return sorted.slice(0, 5);
+  }
+
+  if (sorted && searchType === "results" && sorted.length > 200) {
+    return sorted.slice(0, 200);
+  }
+  return sorted;
+}
+
 /**
  * Searches database for all relevant courses based on query.
- * Returns at most 200 relevant courses based on edit distance.
+ * Returns at most 5 relevant courses in search or 200 relevant courses in results based on edit distance.
  *
  * @param {Search} search: Object that represents the search of a request being passed in.
+ * @param searchType: string that represents a search or result return
  * @returns list of courses if operation was successful, null otherwise.
  */
-export const searchCourses = async ({ search }: SearchQueryType) => {
+export const searchCourses = async ({ search }: SearchQueryType, searchType: string) => {
   try {
     const fullSearch = await fullCourseSearch({ search });
     const sorted = Array.from(fullSearch).sort(courseSort(search.getQuery()));
 
-    if (fullSearch.size > 200 && fullSearch.size > 0) {
-      return sorted.slice(0, 200);
-    }
-
-    return sorted;
+    return courseSlicing(sorted, searchType);
   } catch (e) {
     return null;
   }
 };
 
-export const searchCoursesByProfessor = async ({ search }: SearchQueryType) => {
+export const searchCoursesByProfessor = async ({ search }: SearchQueryType, searchType: string) => {
```

Comment:
Where you add searchType here, which can either be "search" for 5 results or "results" for <200, would it be easier to just migrate to all "search"?